cartridges/int_algolia/cartridge/scripts/algolia/steps/algoliaProductIndex.js

```diff
@@ -71,6 +72,9 @@ exports.beforeStep = function(parameters, stepExecution) {
     ALGOLIA_IN_STOCK_THRESHOLD = algoliaData.getPreference('InStockThreshold');
     INDEX_OUT_OF_STOCK = algoliaData.getPreference('IndexOutOfStock');
 
+    // Try to load `productAttributesConfig.js`, which can be used to override the configs in `algoliaProductConfig.js`.
+    // By default this file does not exist. For an example configuration, see `productAttributesConfig.example.js`.
+    // @TODO: variable name and file name should bear a resemblance so that the file can be looked up by the variable name only -- rename one or the other
```

Comment:
The name of the file is included in the variable name so they bear a resemblance to me. We can rename the variable to `userDefinedProductAttributesConfig` if it makes things clearer but I don't understand the need to find the file with the variable name. In any case it's not worth a TODO comment
```suggestion
```

Reply:
Yes there is a reason: the name of the file is exposed to the users, we want it to be simple.
But in the code, we want an explicit variable name so that it's clear later in the code what the variable contain: some product configuration declared/extended by the users.

The "frustration" is more likely to happen when you have badly named variables.